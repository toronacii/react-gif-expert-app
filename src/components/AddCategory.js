import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [ value, setValue ] = useState('');

  return (
    <form onSubmit={handleSubmit}>
      <input
      type="text"
      value={value}
      onChange={handleChange} />
    </form>
  );

  function handleChange({ target }) {
    setValue(target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    const category = value.trim();
    if (!category) return;

    setCategories(categories => {
      if (!categories.includes(category)) {
        setValue('');
        return [category, ...categories];
      }
      return categories;
    });
  }
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
