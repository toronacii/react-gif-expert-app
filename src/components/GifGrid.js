import PropTypes from 'prop-types';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { gifs, loading } = useFetchGifs(category);

  return <>
    <h3 className="animate__animated animate__flash">{category}</h3>
    { loading && <div className="animate__animated animate__flash">Cargando...</div> }
    <div className="card-grid">
      {gifs.map(({ id, ...props }) => (
        <GifGridItem
          key={id}
          {...props} />)
      )}
    </div>
  </>
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
