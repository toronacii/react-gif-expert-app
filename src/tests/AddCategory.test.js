import { shallow } from 'enzyme';
import React from 'react';
import { AddCategory } from '../components/AddCategory';

describe('Tests on `AddCategory` component', () => {
  const setCategories = jest.fn();
  let wrapper = _createWrapper();

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = _createWrapper();
  });

  test('AddCategory should be created', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should input change', () => {
    const value = 'Dragon ball';
    wrapper.find('input').simulate('change', { target: { value } });

    expect(wrapper.find('input').prop('value')).toBe(value);
  });

  test('shouldn`t submit info when input is empty', () => {
    const form  = wrapper.find('form');
    form.simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should submit info when input isn`t empty', () => {
    const value = 'Dragon ball';

    wrapper.find('input').simulate('change', { target: { value } })
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
  });

  // test('should clean input when form is submited', () => {
  //   const value = 'Dragon ball';

  //   wrapper.find('input').simulate('change', { target: { value } })
  //   wrapper.find('form').simulate('submit', { preventDefault() {} });

  //   expect(wrapper.find('input').prop('value')).toBe('');
  // });

  function _createWrapper() {
    return shallow(<AddCategory setCategories={setCategories} />);
  }
});