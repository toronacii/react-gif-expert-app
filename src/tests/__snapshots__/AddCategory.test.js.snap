// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on \`AddCategory\` component AddCategory should be created 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
